src/e2e/test_harness.ts

```diff
@@ -187,7 +187,8 @@ export function endToEndTestHarness(sdkActions: SDKActions) {
 					token: PERUN_ADDR,
 					asset: new assets.Amount(utils.parseEther("1").toBigInt()),
 				});
-				await sessions[ALICE].transferTo(amount, sessions[BOB].address);
+				sessions[ALICE].transferTo(amount, sessions[BOB].address)
```

Comment:
```
Sent 500 TXs in 0.033s (0.066ms/tx | 15151.515151515152Hz)
```
I wish :laughing: 

Still have to await here first. On the other hand could you confirm that the test times out when using an array of `Promise`s?? It did so on my end and this confuses me a bit... Especially with the updated logic we should not run into issues there.

Reply:
I had another look and it seems like we miss the registration of our hash again. The way we now split the control flow with `.then` chaining does not seem to work reliably. A helperfunction which constructs our `PendingTransaction`s would also be nice to avoid all the code dup when sending TXs. To guarantee sequential execution of `receiptDispatcher.register -> transfer` you probably need a third `Promise` which gets resolved as soon as `register` is done.

```ts
.then((tx) => {
  const hash = tx.hash(...);
  this.receiptDispatcher.register(hash).then(resolveReceiptPromise);
  return this.enclaveWriter.transfer(tx)
}
```

At least this is what I would try, no guarantee. Maybe you come up with a better idea.

Reply:
Just drop `await` to speed up your `js` bro. Trust me bro.

Reply:
Only if you are a certified engineer :clown_face: